{
  ClassNode classNode=new ClassNode("Foo",ACC_PUBLIC,"java.lang.Object");
  classNode.addConstructor(new ConstructorNode(ACC_PUBLIC,null));
  classNode.addProperty(new PropertyNode("bar",ACC_PUBLIC,"java.lang.String","Foo",null,null,null));
  Parameter[] parameters={new Parameter("coll")};
  Statement loopStatement=createPrintlnStatement(new VariableExpression("i"));
  ForStatement statement=new ForStatement("i",new VariableExpression("coll"),loopStatement);
  classNode.addMethod(new MethodNode("iterateDemo",ACC_PUBLIC,"void",parameters,statement));
  Class fooClass=loadClass(classNode);
  assertTrue("Loaded a new class",fooClass != null);
  Object bean=fooClass.newInstance();
  assertTrue("Managed to create bean",bean != null);
  System.out.println("################ Now about to invoke method");
  Object[] array={new Integer(1234),"abc","def"};
  try {
    InvokerHelper.invokeMethod(bean,"iterateDemo",array);
  }
 catch (  InvokerInvocationException e) {
    System.out.println("Caught: " + e.getCause());
    e.getCause().printStackTrace();
    fail("Should not have thrown an exception");
  }
  System.out.println("################ Done");
}
